src/main/java/au/com/outware/cavemanapp/presentation/view/MainActivity.java

```diff
@@ -0,0 +1,144 @@
+    Button deleteButton;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        ButterKnife.bind(this);
+
+        DaggerActivityComponent.builder()
+                .applicationComponent(CavemanApplication.getApplicationComponent())
+                .build()
+                .inject(this);
+
+        listView.setOnItemClickListener(this);
+        listView.setChoiceMode(ListView.CHOICE_MODE_SINGLE);
+
+        // Array Adapter for the list of environments
+        adapter = new ArrayAdapter<>(this,
+                android.R.layout.simple_list_item_single_choice,
+                viewModel.getEnvironments());
+
+        listView.setAdapter(adapter);
+        configureList();
+    }
+
+    private void configureList() {
+        enableButtons(viewModel.shouldEnableButtons());
+        listView.setSelection(viewModel.getSelectedItemPosition());
+        listView.setItemChecked(viewModel.getSelectedItemPosition(), true);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if(requestCode == EditEnvironmentActivity.REQUEST_CODE && resultCode == RESULT_OK) {
+            adapter.clear();
+            adapter.addAll(viewModel.getEnvironments());
+            configureList();
+        }
+    }
+
+    /**
+     * Save the current environment every time the selection is changed
+     */
+    @Override
+    public void onItemClick(AdapterView<?> adapter, View view, int pos, long id) {
+        viewModel.setCurrentEnvironment(this.adapter.getItem(pos));
+        enableButtons(viewModel.shouldEnableButtons());
+    }
+
+    /**
+     * On confirmation delete the environment and save the current environment to production.
+     * Initialize the screen
+     */
+    @Override
+    public void onClick(DialogInterface dialog, int which) {
+        if (which == DialogInterface.BUTTON_POSITIVE) {
+            // delete the environment on confirmation from dialog
+            adapter.remove(viewModel.getCurrentEnvironment());
+            viewModel.removeCurrentEnvironment();
+            configureList();
+        }
+        dialog.dismiss();
+    }
+
+    @OnClick(R.id.btn_add)
+    @SuppressWarnings("unused")
+    void onAddClick() {
+        Intent intent = new Intent(this, EditEnvironmentActivity.class);
+        intent.putExtra(EditEnvironmentActivity.EXTRA_ADD_NEW, true);
+        startActivityForResult(intent, EditEnvironmentActivity.REQUEST_CODE);
+    }
+
+    @OnClick(R.id.btn_edit)
+    @SuppressWarnings("unused")
+    void onEditClick() {
+        startActivity(new Intent(this, EditEnvironmentActivity.class));
+    }
+
+    @OnClick(R.id.btn_delete)
+    @SuppressWarnings("unused")
+    void onDeleteClick() {
+        new AlertDialog.Builder(this).setTitle(R.string.title_delete)
```

Comment:
this dialog could be leaked. We should keep a reference to it and check that it has been dismissed onPause